darts/timeseries.py

```diff
@@ -3,117 +3,554 @@
+
+            # We have to check manually if the index is complete. Another way could be to rely
+            # on `inferred_freq` being present, but this fails for series of length < 3.
+            is_index_complete = len(pd.date_range(self._time_index.min(),
+                                                  self._time_index.max(),
+                                                  freq=self._freq).difference(self._time_index)) == 0
+
+            raise_if_not(is_index_complete, 'Not all timestamps seem to be present in the time index. Does '
+                                            'the series contain holes? If you are using a constructor method, '
+                                            'try specifying `fill_missing_dates=True` '
+                                            'or specify the `freq` parameter.', logger)
+        else:
+            self._freq = 1
+            self._freq_str = None
+
+    """ 
+    Factory Methods
+    ===============
+    """
+
+    @staticmethod
+    def from_xarray(xa: xr.DataArray,
+                    fill_missing_dates: Optional[bool] = True,
+                    freq: Optional[str] = None) -> 'TimeSeries':
+        """
+        Returns a TimeSeries instance built from an xarray DataArray.
+        The dimensions of the DataArray have to be (time, component, sample), in this order. The time
+        dimension can have an arbitrary name, but component and sample must be named "component" and "sample",
+        respectively.
+
+        If two components have the same name or are not strings, this method will disambiguate the components
+        names by appending a suffix of the form "<name>_N" to the N-th column with name "name".
 
         Parameters
         ----------
-        df
-            The actual time series, as a pandas DataFrame with a proper time index.
+        xa
+            The xarray DataArray
+        fill_missing_dates
+            Optionally, a boolean value indicating whether to fill missing dates with NaN values. This requires
+            either a provided `freq` or the possibility to infer the frequency from the provided timestamps.
         freq
-            Optionally, a Pandas offset alias representing the frequency of the DataFrame.
-            (https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases)
-            When creating a TimeSeries instance with a length smaller than 3, this argument must be passed.
-            Furthermore, this argument can be used to override the automatic frequency detection if the
-            index is incomplete.
+            Optionally, a string representing the frequency of the Pandas DataFrame. This is useful in order to fill
+            in missing values if some dates are missing and `fill_missing_dates` is set to `True`.
+
+        Returns
+        -------
+        TimeSeries
+            A univariate or multivariate deterministic TimeSeries constructed from the inputs.
+        """
+
```

Comment:
I think in such cases the constructor is supposed to return an error
```
The time dimension of the DataArray must be indexed either with a DatetimeIndex, or with a RangeIndex.
```
so I would rather rely on that instead of implementing the check twice.